aioredis/util.py

```diff
@@ -319,3 +321,26 @@ def create_future(loop):
 else:
     def create_future(loop):
         return asyncio.Future(loop=loop)
+
+
+make_flat = functools.partial(functools.reduce, operator.add)
+
+
+def make_pairs(*args, **kwargs):
+    if not args and not kwargs:
+        raise TypeError("length of args or kwargs must be > 0")
+
+    elif len(args) == 1 and isinstance(args[0], dict):
+        pairs = make_flat(args[0].items())
+
+    elif len(args) % 2 != 0:
+        raise TypeError("length of args must be even number")
+
+    else:
+        pairs = args
+
+    if kwargs:
+        pairs = make_flat(kwargs.items(), pairs)
```

Comment:
Cool, but what is your kwargs?
```
In [46]: kwargs = dict(enumerate(string.ascii_lowercase))

In [47]: %timeit make_flat(kwargs.items(), tuple())
100000 loops, best of 3: 11.1 µs per loop

In [48]: %timeit make_flat(kwargs.items())
100000 loops, best of 3: 10.3 µs per loop

In [49]: %timeit list(chain.from_iterable(kwargs.items()))
The slowest run took 5.11 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 6.89 µs per loop

In [50]: %timeit list(chain.from_iterable(kwargs.items()))
The slowest run took 5.22 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 6.93 µs per loop

In [51]: %timeit tuple(chain.from_iterable(kwargs.items()))
100000 loops, best of 3: 6.04 µs per loop```